hilti/toolchain/src/compiler/visitors/constant-folder.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020-2021 by the Zeek Project. See LICENSE for details.
+
+#include <hilti/ast/builder/expression.h>
+#include <hilti/ast/ctors/integer.h>
+#include <hilti/ast/detail/visitor.h>
+#include <hilti/ast/expression.h>
+#include <hilti/ast/operators/real.h>
+#include <hilti/ast/operators/signed-integer.h>
+#include <hilti/ast/types/integer.h>
+#include <hilti/base/logger.h>
+#include <hilti/base/util.h>
+#include <hilti/compiler/detail/visitors.h>
+
+using namespace hilti;
+
+namespace {
+
+// For now, this is only a very basic constant folder that only covers cases we
+// need to turn type constructor expressions coming with a single argument into
+// ctor expressions.
+struct VisitorConstantFolder : public visitor::PreOrder<Ctor, VisitorConstantFolder> {
+    // Helper to cast an uint64 to int64, with range check.
+    int64_t to_int64(uint64_t x, position_t& p) {
+        if ( x > std::numeric_limits<int64_t>::max() )
+            logger().error("signed integer value out of range 2", p.node.location());
```

Comment:
Probably 
```suggestion
            logger().error("unsigned integer value out of range", p.node.location());
```

This is untested.

Similary for the two error messages below (these should indeed by `signed integer` though.

Reply:
Turns out this method isn't used anymore, removing.